### PR TITLE
[Compute] `az vm/vmss create`: Add new paramters option `NonPersistedTPM` for `--os-disk-security-encryption-type` to support creating Intel TDX based Confidential VM

### PR DESCRIPTION
**Pull request by Jing-song**
**Related command**

`az vm create/update`

**Description**

Intel TDX based Confidential VM / VMSS Create
close https://github.com/Azure/azure-cli/issues/27479

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -594,6 +594,14 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
             'usage error: The "--os-disk-secure-vm-disk-encryption-set" can only be passed in '
             'when "--os-disk-security-encryption-type" is "DiskWithVMGuestState"')
 
+    if (getattr(namespace, 'os_disk_security_encryption_type', None) and
+            namespace.os_disk_security_encryption_type.lower() == 'nonpersistedtpm'):
+        if (getattr(namespace, 'security_type', None) != 'ConfidentialVM' or
+                getattr(namespace, 'enable_vtpm', None) is not True):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:600]:
```suggestion
        if (getattr(namespace, 'security_type', None) != 'ConfidentialVM') or
                not getattr(namespace, 'enable_vtpm', None):
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -594,6 +594,14 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
             'usage error: The "--os-disk-secure-vm-disk-encryption-set" can only be passed in '
             'when "--os-disk-security-encryption-type" is "DiskWithVMGuestState"')
 
+    if (getattr(namespace, 'os_disk_security_encryption_type', None) and
+            namespace.os_disk_security_encryption_type.lower() == 'nonpersistedtpm'):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:598]:
```suggestion
    os_disk_security_encryption_type = getattr(namespace, 'os_disk_security_encryption_type', None)
    if os_disk_security_encryption_type and os_disk_security_encryption_type.lower() == 'nonpersistedtpm'):
```

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:598]: Left a small comment to make the code more concise

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**